### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by Abbyfa** [Epp-s-Satisfactory-Mods/AutoLink#1] [General]
Tested a blueprint of generators with inputs connected where I placed the lift first and then a splitter last and the coal was going through to the generators just fine. Even after I relogged my game with no mods it continued to work as it did with mods. This should be part of the game, thank you.

**Comment by Epp-code** [comment]
You're welcome!  That makes perfect sense because the mod doesn't change any save game state - it just alters the construction logic to do what I, too, think it should already be doing.  I also just uploaded a new version that should work for more connection types and will make blueprints tileable!

**Comment by iansutherland75** [comment]
This is so fantastic.  I'm not sure how to do the testing, but I have noticed on game reload some connections seemed to unconnect.  The belt would still show next to each other, but would flow would stop completely at the connection point.  These were builds that were previously working right away when I plopped down modular blueprints.  

I'm not sure how to recreate.....

**Reply by Epp-code** [reply]
Thanks for reporting that!  Does this happen consistently and with all your belts or just some of them?  Do you notice anything unique about the belts that don't re-link?  And do you have a save file you wouldn't mind sharing and instructions for where to see the problem?

**Reply by iansutherland75** [reply]
Ack.  I've already fixed it.  If I find it again I'll immediately save.

Will it be a problem that I have other mods installed too?

**Reply by Epp-code** [reply]
Actually, I think I've replicated it and am hunting a fix now.  Thanks again for reporting!

**Reply by Tiken54** [reply]
I noticed the same issue. The solution I found for now is to place a merger/splitter and then remove it. It seems like this action completely resets the belt line, clears everything behind it, and reconnects everything properly.

**Reply by iansutherland75** [reply]
I noticed this mod has gone red in the Mod Manager.  Is it still going forward?

**Reply by Epp-code** [reply]
Yeah, I marked the mod as damaged as a warning that it's not fully functional right now.  I've been working on this for the past two days but it's much trickier than I imagined - without full game source, there's a lot of experimentation and guesswork involved.  I thought I had a solution earlier today but, upon re-testing, it's no longer working so I'm debugging that at the moment.

**Reply by Epp-code** [reply]
I think I just figured it out!  I need to do a bit more testing but I should be able to upload a fixed version tonight.

**Reply by Epp-code** [reply]
Ok, v0.3.0 is uploaded and should fix it (and fix some other logic gaps).  Please test and let me know!

**Reply by Epp-code** [reply]
If you are still monitoring this thread, I just uploaded a 0.4.0 that fixes some issues with rail junctions.

**Reply by Tiken54** [reply]
Perfect, I haven’t had any issues with the belts since version 0.3.0. I’m just starting to work on rail blueprints. The new update comes at the right time

**Reply by Epp-code** [reply]
I just today fixed some more issues I found with railroads so please be sure to get the latest.  Apologies for missing these in testing - to be 100% transparent, I haven't actually made it to trains in the game because I've been working on this mod, so I didn't realize/test all the specifics of how they were used.  I *think* I've got it all covered now but please report any issues.  And if you disassemble and rebuild any blueprint, it should always re-link.
